### PR TITLE
distance fog

### PR DESCRIPTION
**Pull request by greggman**
here's fragment shader side distance fog?

#14601

### CODE REVIEW

File: src/renderers/shaders/ShaderChunk/fog_fragment.glsl [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #ifdef USE_FOG
 
+	float fogDepth = length(fogPosition);
```
Comment (WestLangley) [inline src/renderers/shaders/ShaderChunk/fog_fragment.glsl]: nomenclature and formatting
```glsl
float fogDepth = length( vFogPosition );
```

**Comment by WestLangley** [comment]
`vViewPosition` could be reused when defined, but it is a pain to handle the inclusion of shader chunks as we are currently implementing them. #9035.

So, given that, I think this approach is OK for now.

File: src/renderers/shaders/ShaderChunk/fog_fragment.glsl (hunk 1 above)
Comment (mrdoob) [reply src/renderers/shaders/ShaderChunk/fog_fragment.glsl]: I'll merge and clean up.

**Comment by mrdoob** [comment]
Thanks!

File: src/renderers/shaders/ShaderChunk/fog_fragment.glsl (hunk 1 above)
Comment (mrdoob) [reply src/renderers/shaders/ShaderChunk/fog_fragment.glsl]: The other variables on the file do not use that nomenclature. Will keep it as `fogPosition` for now.

Comment (WestLangley) [reply src/renderers/shaders/ShaderChunk/fog_fragment.glsl]: I know... The implication was to change all occurrences to match the nomenclature convention.

**Comment by mrdoob** [comment]
8 years later we now have fog that doesn't fade away on the edges 🎉
This was specially noticeable in VR...

**Pull request merged**